### PR TITLE
Representation of frames in DebuggingView API.

### PR DESCRIPTION
**Pull request by entlicher**
`DebuggingView` extended with API for frames.

### CODE REVIEW

File: ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java [hunk 1]
```diff
@@ -326,6 +338,37 @@ public final void removePropertyChangeListener(PropertyChangeListener pcl) {
...
+            return 0;
+        }
+
+        /**
+         * Get the stack frames of this thread. Stack frames are provided when the thread
+         * is suspended only.
+         *
+         * @return a list of stack frames, it's empty when the thread is running.
+         */
+        public default List<DVFrame> getFrames() {
```
Comment (JaroslavTulach) [inline ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java:356]: Recently I wrote a [philippic against using default methods](http://wiki.apidesign.org/wiki/BinaryCompatibleDefaultMethods), but given the [one to many proximity](http://wiki.apidesign.org/wiki/Proximity) - e.g. there is very likely just a single implementation of these interfaces in the debugger impl modules - I expect no clashes and `default` methods are the easiest way to extend the interfaces.

File: ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java [hunk 2]
```diff
@@ -401,6 +444,51 @@ public final void removePropertyChangeListener(PropertyChangeListener pcl) {
         public DVThreadGroup[] getThreadGroups();
     }
     
+    /**
+     * Representation of a stack frame in debugging view.
+     * Nodes representing a stack frame in debugging view model should implement this
+     * interface.
```
Comment (JaroslavTulach) [inline ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java:450]: Missing `@since` tag.

**Review by JaroslavTulach — CHANGES_REQUESTED** [review]
I am fine with the concept.

Please add `apichanges.xml`, `@since` tags and increase the spec version of this module. Don't forget to increase dependencies in the modules that are going to use this new API in the future as well.

File: ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java (hunk 2 above)
Comment (entlicher) [reply ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java:454]: Thanks for the reminder of the apichanges, etc. Updated.

File: ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java (hunk 1 above)
Comment (entlicher) [reply ide/spi.debugger.ui/src/org/netbeans/spi/debugger/ui/DebuggingView.java:359]: Right.

**Review by JaroslavTulach — APPROVED** [review]

**Pull request merged**